Tests/TracingTests/ActorTracingTests.swift

```diff
@@ -27,13 +27,17 @@ final class ActorTracingTests: XCTestCase {
 func work() async {}
 
 actor Foo {
-    var bar = 0
+    let bar: LockedValueBox<Int> = .init(0)
```

Comment:
This part is why adding @Sendable to the operation closure is wrong, the withSpan must not be sendable and thus the `bar` does not have to do synchronization -- this would make tracing a complete pain in actors otherwise.

Reply:
Actually, is the answer here to duplicate the `withSpan` methods in `extension Actor`, which implement the methods again using the building blocks of `startAnySpan` etc? That seems to make the warnings go away for me.

Reply:
But why would you want the operation to be sendable? We're always going to invoke it right away and never store it anywhere for "execute somewhere else"; they don't need to be sendable

Reply:
I'm not saying they should be sendable :) I'm trying to fix the warnings I pasted above, and now exploring the idea of the extension on Actor.

Reply:
I'm a bit surprised the extension methods + Sendable does not cause warnings? Either way I think the answer here is just that operation closure just isn't sendable and that's ok 🆗  😄

Reply:
The problem is that this test will then produce warnings when complete concurrency checking is enabled. And client code that tries to do this as well. I think we need _some_ solution here (even if making the closures isn't the right one), the code, as written, produces warnings with strict concurrency checking, today.

Reply:
We're backed into a corner here to be honest. Complete checking is strict, but the language does not offer right tools to solve this. The workaround (that even the stdlib uses) is done via `@_unsafeInheritExecutor` but that's incredibly unsafe and we cannot use it here --  we'd literarily be introducing real dataraces in correct looking code (this attribute removes ALL hop-to operations).

I'd be too worried to put it onto withSpan which is intended to wrap "entire functions" because it'll end up causing actual race conditions at runtime -- by now making any actor hops. 

I think we must wait it out and get to the safe version that would become `@inheritExecutor` and only then would complete checking be correct here... (and in the stdlib as well for that matter).

As potential workaround, I have explored the "overload" on Actor but it also hits a number of problems. We could do this: https://github.com/apple/swift-distributed-tracing/commit/9484f4ba1ad8ca43482666a7a24e3be23692eba9

I talked this over with @FranzBusch and @Lukasa a bunch, and end up coming back to the safe `@inheritExecutor` whenever we hit those issues. This should probably inform the priority of finding a solution for these issues as we plan for enabling complete checking in the near future.

FYI @DougGregor @hborla @tomerd

Reply:
Understood - I'll back out the public API change and we should just land this PR with this warning, and address it separately. I wouldn't want to introduce new public API unless we want to maintain it long term.